crates/live_update/src/jni.rs

```diff
@@ -15,60 +15,73 @@
 use std::ffi::c_void;
 
 use android_logger::Config;
-use figma_import::{fetch_doc, ConvertRequest, Error};
+use jni::objects::{JByteArray, JObject};
 use jni::objects::{JClass, JString};
-use jni::sys::{jbyteArray, jint, JNI_VERSION_1_6};
+use jni::sys::{jint, JNI_VERSION_1_6};
 use jni::{JNIEnv, JavaVM};
 use log::error;
 use log::LevelFilter;
 
-fn throw_basic_exception(env: JNIEnv, msg: String) {
+use figma_import::{fetch_doc, ConvertRequest};
+
+use crate::error_map::map_err_to_exception;
+
+fn throw_basic_exception(env: &mut JNIEnv, msg: String) {
     //An error occurring while trying to throw an exception shouldn't happen, but let's at least panic with a decent error message
     env.throw(msg).expect("Error while trying to throw Exception");
 }
 
-fn jni_fetch_doc(
-    env: JNIEnv,
+fn jni_fetch_doc<'local>(
```

Comment:
What changed such that you need this lifetime specifier now? I don't get how that works since this is called from Kotlin isn't it?

Reply:
Part of the upgrade to jni-rs 0.21: https://github.com/jni-rs/jni-rs/blob/master/docs/0.21-MIGRATION.md

"Most of these changes are needed to ensure that all local references (JObject and the like) have the correct lifetime and can't be used after they're deleted, which would cause undefined behavior. See https://github.com/jni-rs/jni-rs/issues/392 for a discussion of the problem these changes solve."